ql/src/java/org/apache/hadoop/hive/ql/ddl/table/create/CreateTableOperation.java

```diff
@@ -53,8 +59,33 @@ public CreateTableOperation(DDLOperationContext context, CreateTableDesc desc) {
     super(context, desc);
   }
 
+  private void readSchemaFromFile() throws HiveException {
+    String fileFormat = desc.getLikeFileFormat();
+    StorageFormatFactory storageFormatFactory = new StorageFormatFactory();
+    StorageFormatDescriptor descriptor = storageFormatFactory.get(fileFormat);
+    if (descriptor == null) {
+      // normal operation should never hit this since analysis has already verified this exists
```

Comment:
The comment here seems to indicate this should be an assert rather than an exception.